### ISSUE TITLE
Allow --edit even when the urls.yaml file is currently invalid

### ISSUE THREAD

**Issue by kbabioch** [thp/urlwatch#247]
Once there is a configuration error the `--edit`, `--edit-config` and `--edit-hooks` commands no longer work.

For example if the `urls.yaml` looks like this:

    name: test
    ur: abc

There will be an error when invoking the `--edit` command:

    EDITOR=vim urlwatch --edit
    Traceback (most recent call last):
      File "/usr/bin/urlwatch", line 107, in <module>
        urlwatch = Urlwatch(command_config, config_storage, cache_storage, urls_storage)
      File "/usr/lib/python3.6/site-packages/urlwatch/main.py", line 63, in __init__
        self.load_jobs()
      File "/usr/lib/python3.6/site-packages/urlwatch/main.py", line 84, in load_jobs
        jobs = self.urls_storage.load_secure()
      File "/usr/lib/python3.6/site-packages/urlwatch/storage.py", line 233, in load_secure
        jobs = self.load()
      File "/usr/lib/python3.6/site-packages/urlwatch/storage.py", line 322, in load
        return [JobBase.unserialize(job) for job in yaml.load_all(fp) if job is not None]
      File "/usr/lib/python3.6/site-packages/urlwatch/storage.py", line 322, in <listcomp>
        return [JobBase.unserialize(job) for job in yaml.load_all(fp) if job is not None]
      File "/usr/lib/python3.6/site-packages/urlwatch/jobs.py", line 118, in unserialize
        raise ValueError('Kind is not specified, and no job matches: %r' % (data,))
    ValueError: Kind is not specified, and no job matches: {'name': 'test', 'ur': 'abc'}

urlwatch should not parse the config when trying to edit it. Once misconfigured, it cannot be fixed without editing the files directly. There should also be a basic syntax check when saving the file through the `--edit` command. The same is true for any other configuration file (hooks.py, urlwatch.yaml).

**Comment by thp** [comment]
This is actually how it works if you are just using `--edit` for editing the config:

https://github.com/thp/urlwatch/blob/master/lib/urlwatch/storage.py#L174

It will only save the file if the syntax check passes after the edit is complete, if your edit results don't parse, then the results won't be saved, and it will tell you where your modified (but broken) file is so you can recover if you need to.

Of course, once the file is broken (e.g. by editing manually outside) then there's no way to really recover from that except by fixing the error manually (but if you have manually edited the file before, then you should know how to manually edit it afterwards).

In other words, there's no way to make `urls.yaml` be the way you mention above apart from manual editing, and once you're manually editing the file anyway, you might just as well manually edit it to fix it. The `--edit` would take care of that if you use it all the time.

**Comment by kbabioch** [comment]
> Of course, once the file is broken (e.g. by editing manually outside) then there's no way to really recover from that except by fixing the error manually (but if you have manually edited the file before, then you should know how to manually edit it afterwards).

Hm, then I probably edited by hand and forgot about it :-).

Nevertheless, I think that `--edit` should always open the file in the editor and only complain when trying to save it, not before actually opening it.

**Comment by thp** [comment]
> Nevertheless, I think that --edit should always open the file in the editor and only complain when trying to save it, not before actually opening it.

Yes, that part makes sense.

**Comment by kbabioch** [comment]
By the way: This part does not work as previously described by you.

I'm using the latest release in a virtual environment:

    $ urlwatch --version
    urlwatch 2.13

And removed any configuration and cache files:

    $ rm -r ~/.config/urlwatch ~/.cache/urlwatch

Then I try to edit the URL file. urlwatch will create a new / empty one:

    $ ./venv/bin/urlwatch --edit                

This will open a new (empty) file in the editor.  When I will it with this content:

    url2: a

It is successfully saved:

    $ ./venv/bin/urlwatch --edit                

        A default config has been written to /home/kbabioch/.config/urlwatch/urlwatch.yaml.
        Use "urlwatch --edit-config" to customize it.
            
    No jobs file found
    Saving edit changes in /home/kbabioch/.config/urlwatch/urls.yaml

Now, when I try to edit it once again, this happens:

    $ ./venv/bin/urlwatch --edit
    Traceback (most recent call last):
      File "./venv/bin/urlwatch", line 6, in <module>
        exec(compile(open(__file__).read(), __file__, 'exec'))
      File "/home/kbabioch/urlwatch/urlwatch", line 107, in <module>
        urlwatch = Urlwatch(command_config, config_storage, cache_storage, urls_storage)
      File "/home/kbabioch/urlwatch/lib/urlwatch/main.py", line 63, in __init__
        self.load_jobs()
      File "/home/kbabioch/urlwatch/lib/urlwatch/main.py", line 84, in load_jobs
        jobs = self.urls_storage.load_secure()
      File "/home/kbabioch/urlwatch/lib/urlwatch/storage.py", line 233, in load_secure
        jobs = self.load()
      File "/home/kbabioch/urlwatch/lib/urlwatch/storage.py", line 322, in load
        return [JobBase.unserialize(job) for job in yaml.load_all(fp) if job is not None]
      File "/home/kbabioch/urlwatch/lib/urlwatch/storage.py", line 322, in <listcomp>
        return [JobBase.unserialize(job) for job in yaml.load_all(fp) if job is not None]
      File "/home/kbabioch/urlwatch/lib/urlwatch/jobs.py", line 118, in unserialize
        raise ValueError('Kind is not specified, and no job matches: %r' % (data,))
    ValueError: Kind is not specified, and no job matches: {'url2': 'a'}
        
As you can see I never touched the file manually, it was saved successfully, although it is invalid.

By the way: I'm not sure if this is known / wanted. When installing the package in a virtual environment via pip, there is no template when editing the file, e.g. the file openend when invoking `--edit` is empty. On a system installation (Arch Linux), the file from `/usr/share/urlwatch/examples/urls.yaml.example` is copied into place, which contains some examples, etc.
